testing/tests/benchmark/test_testing.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmark tests for ops-scenario."""
+
+import dataclasses
+import pathlib
+import sys
+
+import ops
+from ops import testing
+
+sys.path.append(
+    str(
+        pathlib.Path(__file__).parent.parent.parent.parent
+        / "test"
+        / "charms"
+        / "test_benchmark"
+        / "src"
+    )
+)
```

Comment:
WDYT about one of these?

```suggestion
sys.path.append(str(pathlib.Path(__file__) / "../../../../test/charms/test_benchmark/src"))
```

or 

```suggestion
sys.path.append(str(pathlib.Path(__file__).parent / "../../../test/charms/test_benchmark/src"))
```

if a construct like `sub/dir/file.py/../../../another/dir` is too jarring...

I'm a little wary of changing import path at import time, as this could hypothetically affect other tests.

Alternatives would include:
- path tweak and import in a fixture, or
- explicit path setup in pytest config

Reply:
Agreed, it would be better to minimise the impact. I've moved it to a fixture, thanks.